state-chain/pallets/cf-ingress-egress/src/migrations/deposit_channels_with_boost_fee.rs

```diff
@@ -0,0 +1,217 @@
+	fn on_runtime_upgrade() -> Weight {
+		DepositChannelLookup::<T, Instance2>::translate(
+			|_address: PolkadotAccountId, old_channel: old::DepositChannelDetails<T, Instance2>| {
+				Some(DepositChannelDetails::<T, Instance2> {
+					deposit_channel: old_channel.deposit_channel,
+					opened_at: old_channel.opened_at,
+					expires_at: old_channel.expires_at,
+					action: old_channel.action,
+					boost_fee: 0,
+				})
+			},
+		);
+		Weight::zero()
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+		let number_of_channels_in_lookup =
+			old::DepositChannelLookup::<T, Instance2>::iter_keys().count() as u32;
+
+		Ok(number_of_channels_in_lookup.encode())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(state: Vec<u8>) -> Result<(), DispatchError> {
+		let number_of_channels_in_lookup_pre_migration = <u32>::decode(&mut &state[..]).unwrap();
+		ensure!(
+			DepositChannelLookup::<T, Instance2>::iter_keys().count() as u32 ==
+				number_of_channels_in_lookup_pre_migration,
+			"DepositChannelLookup migration failed."
+		);
+		Ok(())
+	}
+}
+
+impl<T: Config<Instance3, TargetChain = Bitcoin>> OnRuntimeUpgrade for Migration<T, Instance3> {
+	fn on_runtime_upgrade() -> Weight {
+		DepositChannelLookup::<T, Instance3>::translate(
+			|_address: ScriptPubkey, old_channel: old::DepositChannelDetails<T, Instance3>| {
+				Some(DepositChannelDetails::<T, Instance3> {
+					deposit_channel: old_channel.deposit_channel,
+					opened_at: old_channel.opened_at,
+					expires_at: old_channel.expires_at,
+					action: old_channel.action,
+					boost_fee: 0,
+				})
+			},
+		);
+		Weight::zero()
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+		let number_of_channels_in_lookup =
+			old::DepositChannelLookup::<T, Instance3>::iter_keys().count() as u32;
+
+		Ok(number_of_channels_in_lookup.encode())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(state: Vec<u8>) -> Result<(), DispatchError> {
+		let number_of_channels_in_lookup_pre_migration = <u32>::decode(&mut &state[..]).unwrap();
+		ensure!(
+			DepositChannelLookup::<T, Instance3>::iter_keys().count() as u32 ==
+				number_of_channels_in_lookup_pre_migration,
+			"DepositChannelLookup migration failed."
+		);
+		Ok(())
+	}
+}
+
+#[cfg(test)]
+mod migration_tests {
+	use cf_chains::btc::{
+		deposit_address::{DepositAddress, TapscriptPath},
+		BitcoinScript, ScriptPubkey,
+	};
+
+	use self::mock_btc::new_test_ext;
+
+	use super::*;
+	use crate::mock_btc::*;
+
+	#[test]
+	fn test_migration() {
+		new_test_ext().execute_with(|| {
+			let address1 = ScriptPubkey::Taproot([0u8; 32]);
+			let address2 = ScriptPubkey::Taproot([1u8; 32]);
+
+			// Insert mock data into old storage
+			old::DepositChannelLookup::insert(
+				address1.clone(),
+				old::DepositChannelDetails::<Test, _> {
+					deposit_channel: DepositChannel {
+						channel_id: 123,
+						address: address2.clone(),
```

Comment:
address 1 has address 2 in it?

I would just pull this insert into a helper function to hide these unimportant details (and help ensure they're correct/reasonable